Lib/profiling/sampling/live_collector.py

```diff
@@ -0,0 +1,2005 @@
+                col += 12
+
+            # Function name column
+            if col < width - 15:
+                remaining_space = width - col - 1
+                func_width = min(
+                    MAX_FUNC_NAME_WIDTH,
+                    max(MIN_FUNC_NAME_WIDTH, remaining_space // 2),
+                )
+
+                func_display = funcname
+                if len(funcname) > func_width:
+                    func_display = funcname[: func_width - 3] + "..."
+                func_display = f"{func_display:<{func_width}}"
+                self.add_str(line, col, func_display, color_func)
+                col += func_width + 2
+
+                # File:line column
+                if col < width - 10:
+                    simplified_path = self.collector._simplify_path(filename)
+                    file_line = f"{simplified_path}:{lineno}"
+                    remaining_width = width - col - 1
+                    self.add_str(
+                        line, col, file_line[:remaining_width], color_file
+                    )
+
+            line += 1
+
+        return line
+
+
+class FooterWidget(Widget):
+    """Widget for rendering the footer section (legend and controls)."""
+
+    def __init__(self, display, colors, collector):
+        """
+        Initialize footer widget.
+
+        Args:
+            display: DisplayInterface implementation
+            colors: Dictionary of color attributes
+            collector: Reference to LiveStatsCollector for accessing state
+        """
+        super().__init__(display, colors)
+        self.collector = collector
+
+    def render(self, line, width, **kwargs):
+        """
+        Render the footer at the specified position.
+
+        Args:
+            line: Starting line number (should be height - 2)
+            width: Available width
+
+        Returns:
+            Next available line number
+        """
+        A_DIM = self.display.get_attr("A_DIM")
+        A_BOLD = self.display.get_attr("A_BOLD")
+
+        # Legend line
+        legend = "nsamples: direct/cumulative (direct=executing, cumulative=on stack)"
+        self.add_str(line, 0, legend[: width - 1], A_DIM)
+        line += 1
+
+        # Controls line with status
+        sort_names = {
+            "tottime": "Total Time",
+            "nsamples": "Direct Samples",
+            "cumtime": "Cumulative Time",
+            "sample_pct": "Sample %",
+            "cumul_pct": "Cumulative %",
+        }
+        sort_display = sort_names.get(
+            self.collector.sort_by, self.collector.sort_by
+        )
+
+        # Build status indicators
+        status = []
+        if self.collector.finished:
+            status.append("[PROFILING FINISHED - Press 'q' to quit]")
+        elif self.collector.paused:
+            status.append("[PAUSED]")
+        if self.collector.filter_pattern:
+            status.append(f"[Filter: {self.collector.filter_pattern}]")
```

Comment:
What do you think about adding the note on how to clear it to make it easier to find the keyboard shortcut?
```suggestion
            status.append(f"[Filter: {self.collector.filter_pattern} (c to clear)]")
```